presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -377,6 +378,12 @@ public TaskInfo updateTask(
             // TODO: This should have been done when the QueryContext was created. However, the session isn't available at that point.
             queryContexts.getUnchecked(taskId.getQueryId()).setResourceOvercommit();
         }
+        else {
+            queryContexts.getUnchecked(
+                    taskId.getQueryId()).setMemoryLimits(
```

Comment:
Yeah I noticed that pain point. It took a while to arrive at `updateTask` and see that I was not the first person to have this issue.

I didn't want to change it because I assumed the design is such that it can be retrieved before `updateTask` is called for a reason. All you need is a taskId.